vars/pullGitRepositoryToSapCloudPlatformAbapEnvironment.groovy

```diff
@@ -0,0 +1,213 @@
+import static com.sap.piper.Prerequisites.checkScript
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.JenkinsUtils
+import com.sap.piper.Utils
+import groovy.json.JsonSlurper
+import hudson.AbortException
+import groovy.transform.Field
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+import java.util.UUID
+
+@Field def STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = [
+    /**
+     * Specifies the host address of the SAP Cloud Platform ABAP Environment system
+     */
+    'host',
+    /**
+     * Specifies the name of the Repository (Software Component) on the SAP Cloud Platform ABAP Environment system
+     */
+    'repositoryName'
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus([
+    /**
+     * Specifies the communication user of the communication scenario SAP_COM_0510
+     */
+    'username',
+    /**
+     * Specifies the password of the communication user
+     */
+    'password'
+])
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+/**
+ * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
+ *
+ * !!! note "Git Repository and Software Component"
+ *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
+ *       Currently, those two names are used synonymous.
+ * !!! note "User and Password"
+ *        In the future, we want to support the user / password creation via the create-service-key funcion of cloud foundry.
+ *        For this case, it is not possible to use the usual pattern with Jenkins Credentials.
+ */
+@GenerateDocumentation
+void call(Map parameters = [:]) {
+
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
+
+        def script = checkScript(this, parameters) ?: this
+
+        // In the future, we want to support the user / password creation via the create-service-key funcion of cloud foundry.
+        // For this case, it is not possible to use the usual pattern with Jenkins Credentials.
+        Map configuration = ConfigurationHelper.newInstance(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('host', 'Host not provided')
+            .withMandatoryProperty('repositoryName', 'Repository / Software Component not provided')
+            .withMandatoryProperty('username')
+            .withMandatoryProperty('password')
+            .collectValidationFailures()
```

Comment:
`collectValidationFailures` must be set before the calls to `collectValidationFailures` in order to get all the failed parameters at once instead of one after the other.

Reply:
done